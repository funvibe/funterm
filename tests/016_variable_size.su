lua.packet = <<5:8, "Hello":5/binary, " World">>

match lua.packet {
    <<size:8, data:size/binary, rest/binary>> -> {
        lua.print("Size:", size)
        lua.print("Data:", data)
        lua.print("Rest:", rest)
    }
}

# Более сложный пример с вычислением размера
lua.complex = <<10:8, "DATA":4/binary, "EXTRA":5/binary, "END">>

match lua.complex {
    <<total:8, payload:(total-6)/binary, trailer/binary>> -> {
        lua.print("Total size:", total)
        lua.print("Payload:", payload)
        lua.print("Trailer:", trailer)
    }
}