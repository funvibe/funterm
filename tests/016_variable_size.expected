Size: 5
Data: Hello
Rest:  World
Total size: 10
Payload: DATA
Trailer: EXTRAEND